src/java/org/apache/cassandra/utils/SystemInfo.java

```diff
@@ -0,0 +1,211 @@
+     */
+    long getVirtualMemoryMax() {
+        return si.getOperatingSystem().getCurrentProcess().getVirtualSize();
+    }
+
+    /**
+     *
+     * @return The amount of swap space allocated on the system.
+     */
+    long getSwapSize() {
+        return si.getHardware().getMemory().getVirtualMemory().getSwapTotal();
+    }
+
+    /**
+     * @return the PID of the current system.
+     */
+    public long getPid() {
+        return si.getOperatingSystem().getProcessId();
+    }
+
+    /**
+     * Tests if the system is running in degraded mode.
+     * If the system is running in degraded mode this method will log information.
+     * @return @{code true} if the system is in degraded mode, @{code false} otherwise.
+     */
+    public Optional<String> isDegraded()
+    {
+        Supplier<Optional<String>> expectedNumProc = () -> {
+            // only check proc on nproc linux
+            if (oshi.SystemInfo.getCurrentPlatform() == PlatformEnum.LINUX)
+                return invalid(getMaxProcess(), EXPECTED_NPROC) ? of(format("Number of processes should be >= %s. ", EXPECTED_NPROC))
+                                                                : empty();
+            else
+                return of(format("System is running %s, Linux OS is recommended", platform()));
+        };
+
+        Supplier<Optional<String>> swapShouldBeDisabled = () -> (getSwapSize() > 0)
+                                                                ? of("Swap should be disabled. ")
+                                                                : empty();
+
+        Supplier<Optional<String>> expectedAddressSpace = () -> invalid(getVirtualMemoryMax(), EXPECTED_AS)
+                                                                ? of(format("Amount of available address space should be >= %s. ", EXPECTED_AS))
+                                                                : empty();
+
+        Supplier<Optional<String>> expectedMinNoFile = () -> invalid(getMaxOpenFiles(), EXPECTED_MIN_NOFILE)
+                                                             ? of(format("Minimum value for max open files should be >= %s. ", EXPECTED_MIN_NOFILE))
+                                                             : empty();
+
+        StringBuilder sb = new StringBuilder();
+
+        for (Supplier<Optional<String>> check : ImmutableList.of(expectedNumProc,
+                                                                 swapShouldBeDisabled,
+                                                                 expectedAddressSpace,
+                                                                 expectedMinNoFile))
+        {
+            check.get().map(sb::append);
+        }
+
+        String message = sb.toString();
+        return message.isEmpty() ? empty() : of(message);
+    }
+
+    /**
+     * Checks if a value if valide (i.e. value >= min or value == INFINITY..
```

Comment:
@Claudenw there is still messed up end of this line. Should not it end with `INFINITY)` ?